app/tests/infrastructure/test_exchange_repository.py

```diff
@@ -0,0 +1,52 @@
+from unittest.mock import patch
+
+import pytest
+
+from hallebarde.infrastructure import exchange_repository
+
+
+TABLE_NAME = f'hallebarde-dev-table'
+
+
+@pytest.mark.usefixtures("get_dynamodb_table")
+class TestExchangeRepository:
+
+    @patch('hallebarde.infrastructure.exchange_repository._get_dynamodb_table')
+    def test_save_should_create_readable_item(self, mock_get_table, generic_exchange, get_dynamodb_table):
+        # Given
+        mock_get_table.return_value = get_dynamodb_table
+
+        # When
+        exchange_repository.save(generic_exchange)
+
+        # Then
+        actual_exchange = exchange_repository.get('e68faee7-753e-49a6-a372-dc1cdc0dae03')
+        assert actual_exchange == generic_exchange
+
+    @patch('hallebarde.infrastructure.exchange_repository._get_dynamodb_table')
+    def test_get_should_return_none_if_no_id_is_matched(self, mock_get_table, generic_exchange, get_dynamodb_table):
+        # Given
+        mock_get_table.return_value = get_dynamodb_table
+
+        # When
+        exchange_repository.save(generic_exchange)
+
+        # Then
+        actual_exchange = exchange_repository.get('non-existent-id')
+        assert actual_exchange is None
+
+    @patch('hallebarde.infrastructure.exchange_repository._get_dynamodb_table')
+    def test_get_account_exchanges_should_return_only_account_exchanges(self, mock_get_table,
+                                                                        exchanges_with_last_one_with_different_emails,
+                                                                        get_dynamodb_table):
+        # Given
+        mock_get_table.return_value = get_dynamodb_table
+        for exchange in exchanges_with_last_one_with_different_emails:
+            exchange_repository.save(exchange)
+
+        # When
+        actual_exchanges = exchange_repository.get_account_exchanges(exchanges_with_last_one_with_different_emails[0].email)
```

Comment:
I find the test difficult to read with this `exchanges_with_last_one_with_different_emails`.
Maybe we should split this in three exchange fixtures

Reply:
Agreed, I will fix the code according to that